tools/iree-prof-output-stdout.cc

```diff
@@ -57,37 +58,30 @@ std::string GetDurationStr(int64_t duration_ns,
   }
 }
 
-// Returns the duration per thread, i.e from the earliest start to the latest
-// end of zones running on each thread.
+// Returns the duration per thread, i.e merged durations of all zones running on
+// each thread.
 template <typename T>
 absl::flat_hash_map<int, int64_t> GetThreadDurations(
     const tracy::Worker& worker,
     const tracy::unordered_flat_map<int16_t, T>& zones,
     const std::vector<std::string>& thread_substrs) {
```

Comment:
I think having substrings as a filter option made the logic more complicated.

We have the following items:

1. Total execution time.
2. Memory transfer & dispatch times per thread

I think the entities from 2) are distinct and they are not supposed to overlap.

Just from the above, we can have
1. accumulated dispatch and memory transfer times for each thread
2. averaged dispatch and memory transfer times time (for all threads)

Can the assumption with no overlapping simply the logic?

Reply:
Changed to get durations only from top-level zones which are listed in the timeline in thread data. It's just simple summation, not comparing start/end of zones.